wallet/wallet.go

```
@@ -240,9 +268,51 @@ func NewWalletWithConfig(config Config) (*Wallet, error) {
+			return fmt.Errorf("client authentication key is not compatible with ES256: %w", err)
+		}
+		var publicKey *ecdsa.PublicKey
+
+		switch k := config.Key.PublicKey().Key.(type) {
+		case *ecdsa.PublicKey:
+			publicKey = k
+		case ecdsa.PublicKey:
+			publicKey = &k
+		}
+
+		if publicKey == nil || publicKey.Curve == nil || publicKey.Curve.Params() == nil ||
```

golangci-lint (2.12.2)

[error] QF1008: could remove embedded field "Curve" from selector

(staticcheck)